src/klein/test/test_headers_compat.py

```diff
@@ -0,0 +1,66 @@
+# -*- test-case-name: klein.test.test_headers -*-
+# Copyright (c) 2017-2018. See LICENSE for details.
+
+"""
+Tests for L{klein._headers}.
+"""
+
+from twisted.web.http_headers import Headers
+
+from ._trial import TestCase
+from .test_headers import MutableHTTPHeadersTestsMixIn
+from .._headers import (
+    IMutableHTTPHeaders, RawHeaders, normalizeRawHeadersFrozen
+)
+from .._headers_compat import HTTPHeadersWrappingHeaders
+
+# Silence linter
+IMutableHTTPHeaders, RawHeaders
+
+
+__all__ = ()
+
+
+
+class HTTPHeadersWrappingHeadersTests(MutableHTTPHeadersTestsMixIn, TestCase):
```

Comment:
I...any chance you can turn this into a composition rather than inheritance? This seems unnecessary to mix-in.

Reply:
I've thought about that but I think the best way to do that is to fix trial to provide some means for composition instead of having to subclass `*TestCase`.  I'm not really ready to dive into that problem.

The biggest argument against subclassing (and especially mix-ins) is, I think, that it's stupidly hard to follow the intentions and track down the parent class, but it's all in one file, and test cases are not public API… So I think in examples like this one, it seems academically interesting but ultimately a bunch of work for negligible gain.